app/chain_api.py

```diff
@@ -30,6 +30,10 @@ def __init__(self, services):
         await self.auth_svc.check_permissions(request)
         data = dict(await request.json())
         index = data.pop('index')
+        try:
+            refresh_type = data.pop('refresh_key')
```

Comment:
I don't understand the refresh_key.. can you explain?

Reply:
refresh_key is a notification flag set by the web UI when trying to pull operation information, and it only exists if the endpoint is called from the web UI (or with intent). I did it this way because the pulling of that information takes a little bit of time, so we only want to do it if absolutely necessary, and the same general selection parameters used to select the operation data also apply to another call at some point (I think it's the rats checking in for data, but I could be mistaken).

Reply:
ok I see what you're aiming for. good thought-- but I'd rather see these two things split:

1. getting of technique information when explode_operation is called
2. handling the chain_api calls to be more efficient. 

The former should be straightforward but I agree that the explode_operation function is getting a bit heavy, so i wouldn't mind a performance boost there. I would focus the performance boost on making each piece more asynchronous (we currently get adversaries, abilities, links on the chain (core_chain), facts, etc. I imagine we can do a lot of these actions at the same time... 

Alternatively, another option I've been interested in is using the "in-memory" SQLite option when an operation starts and flush the contents to the database file at the end of the operation (or every minute or something). Since the only difference between the in-memory and file databases is how they are instantiated, this should be a pretty small effort-- but could give us a really optimal way of handling the data during an operation.

Reply:
I've tweaked things slightly so that it uses a secondary endpoint to grab the extra necessary data. Let me know if you don't like this approach, but it keeps things internal to a single plugin rather than splitting it up, and avoids adding additional flags, while limiting the additional data to only where it's needed.

Reply:
I agree with making things more efficient, but that's a completely separate task/challenge.